src/benchq/resource_estimation/graph/extrapolation_estimator.py

```diff
@@ -84,64 +85,64 @@ def _get_extrapolated_graph_data(
         self,
+        algorithm_description: AlgorithmDescription,
         data: List[ResourceInfo],
-        error_budget: ErrorBudget,
-        delayed_gate_synthesis: bool,
-        steps_to_extrapolate_to: int,
     ):
+        assert isinstance(algorithm_description.program, QuantumProgram)
         extrapolated_info = self._get_extrapolated_graph_data(
-            data, steps_to_extrapolate_to
+            data, algorithm_description.program
         )
         resource_info = self._estimate_resources_from_graph_data(
-            extrapolated_info, delayed_gate_synthesis, error_budget
+            extrapolated_info, algorithm_description
         )
         return ExtrapolatedResourceInfo(
             n_logical_qubits=resource_info.n_logical_qubits,
             n_measurement_steps=resource_info.n_measurement_steps,
-            n_nodes=resource_info.n_nodes,
-            synthesis_multiplier=resource_info.synthesis_multiplier,
+            n_nodes=algorithm_description.program.n_nodes,
+            n_t_gates=resource_info.n_t_gates,
+            n_rotation_gates=resource_info.n_rotation_gates,
             code_distance=resource_info.code_distance,
             logical_error_rate=resource_info.logical_error_rate,
-            total_time=resource_info.total_time,
+            total_time_in_seconds=resource_info.total_time_in_seconds,
             n_physical_qubits=resource_info.n_physical_qubits,
             decoder_power=resource_info.decoder_power,
             decoder_area=resource_info.decoder_area,
             max_decodable_distance=resource_info.max_decodable_distance,
             n_logical_qubits_r_squared=extrapolated_info.max_graph_degree_r_squared,
             n_measurement_steps_r_squared=extrapolated_info.n_measurement_steps_r_squared,  # noqa: E501
-            n_nodes_r_squared=extrapolated_info.n_nodes_r_squared,
             data_used_to_extrapolate=data,
-            steps_to_extrapolate_to=steps_to_extrapolate_to,
+            steps_to_extrapolate_to=algorithm_description.program.steps,
         )
 
 
 def _get_linear_extrapolation(x, y, steps_to_extrapolate_to):
     coeffs, sum_of_residuals, _, _, _ = np.polyfit(x, y, 1, full=True)
     r_squared = 1 - (sum_of_residuals[0] / (len(y) * np.var(y)))
     m, c = coeffs
-    return ceil(m * steps_to_extrapolate_to + c), r_squared
+
+    # get rid of floating point errors
+    rounded_point = round(m * steps_to_extrapolate_to + c, 5)
+    return ceil(rounded_point), r_squared
 
 
 def _get_logarithmic_extrapolation(x, y, steps_to_extrapolate_to):
     log_x = np.log(x)
     coeffs, sum_of_residuals, _, _, _ = np.polyfit(log_x, y, 1, full=True)
     r_squared = 1 - (sum_of_residuals[0] / (len(y) * np.var(y)))
     m, c = coeffs
-    return ceil(m * np.log(steps_to_extrapolate_to) + c), r_squared
+
+    # get rid of floating point errors
+    rounded_point = round(m * np.log(steps_to_extrapolate_to) + c, 5)
+    return ceil(rounded_point), r_squared
```

Comment:
We don't technically need to round here. But I think it makes the most sense to give back an integer here since the variable should be an integer anyways.

Reply:
My question was probably a bit imprecise. What I meant is "Why do we call `round` with 5 digits if we put the result into `ceil`, which effectively annihilates the rounding?"

Reply:
Oh! That's a mistake. Will fix.